sky/clouds/vast.py

```diff
@@ -255,15 +282,15 @@ def _make(instance_list):
         assert len(accelerators) == 1, resources
         acc, acc_count = list(accelerators.items())[0]
         (instance_list,
-         fuzzy_candidate_list) = catalog.get_instance_type_for_accelerator(
+         fuzzy_candidate_list) = vast_catalog.get_instance_type_for_accelerator(
              acc,
              acc_count,
              use_spot=resources.use_spot,
              cpus=resources.cpus,
              region=resources.region,
              zone=resources.zone,
              memory=resources.memory,
-             clouds='vast')
+             datacenter_only=datacenter_only)
```

Comment:
it seems the datacenter_only is only used for the catalog filtering. How about we don't change the code in this file, but just query the config in the `vast_catalog` file underthehood?